corehq/apps/case_search/utils.py

```diff
@@ -102,4 +105,17 @@ def _add_case_property_queries(self):
             for removal_regex in remove_char_regexs:
                 to_remove = re.escape(removal_regex.regex)
                 value = re.sub(to_remove, '', value)
-            self.search_es = self.search_es.case_property_query(key, value, fuzzy=(key in fuzzies))
+            if self.wildcard_enabled:
+                values = value.split()
+                if len(values) == 1 and key in fuzzies:
+                    # fuzzy search can only be supported for single words
+                    self.search_es = self.search_es.case_property_query(
+                        key, value, fuzzy=True, clause=queries.SHOULD)
+                    clause = queries.SHOULD
+                else:
+                    clause = queries.MUST
+                for token in values:
```

Comment:
If I understand this correctly this adds `.*` before and after each character in the search string? That seems like it is going to 1) be extremely inefficient and 2) return totally random results. I haven't looked into this, but is this the standard way of doing this type of search with elasticsearch? There might be something you can do with the tokenizer to better support this workflow in a more efficient manner.

Reply:
Hey Farid, this is something that I am already aware of and hence flagged the potential performance issue in the PR description as well. There are two approaches possible 1. The way I am doing via regex 2. By using ngram tokenizer in the index creation. Using ngram tokenizer also has the penalty of much larger index sizes, and also going on that route means reindex of the case-search index for all the domains, whereas only there might be just one domain that needs this feature. For these reasons, I am first trying approach 1, which is much simpler to implement and test. I have already tested these queries on a load-heavy domain on staging which went without any issues. So far, I haven't found any performance issues either in the query times or in the load on the ES cluster. Since this is on a feature flag, we will be able to test on production with gradually increasing case-load domains. Given the results on staging, I am fairly optimistic that this is not going to be an issue on Production, especially since there are very small number of domains that use the case-search index. If we are going to hit any bottlenecks, I am planning to take the ngram approach.

Reply:
OK, sounds like you've thought through this already! Thanks, I'm reassured by your process, and agree that the fact it is behind a feature flag means you have a lever to turn it off without having to redeploy.